packages/cosmic-swingset/Makefile

```diff
@@ -66,36 +66,45 @@ scenario1-run-client:
 	AG_SOLO_BASEDIR=t7 ve3/bin/ag-setup-solo --webhost=127.0.0.1:$(BASE_PORT)
 
 AGC = ./lib/ag-chain-cosmos
-scenario2-setup: build-cosmos scenario2-setup-nobuild
+AGCH = ag-cosmos-helper
+scenario2-setup: all scenario2-setup-nobuild
 scenario2-setup-nobuild:
-	rm -rf ~/.ag-chain-cosmos
-	$(AGC) init scenario2-chain --chain-id=$(CHAIN_ID)
 	rm -rf t1
 	mkdir t1
+	# Init the chain node.
+	$(AGC) --home=t1/n0 init scenario2-chain --chain-id=$(CHAIN_ID)
+	# Init all the ag-solos.
 	set -e; for port in `seq $(BASE_PORT) $$(($(BASE_PORT) + $(NUM_SOLOS) - 1))`; do \
 		bin/ag-solo init t1/$$port --webport=$$port; \
-		case $$port in \
-			$(BASE_PORT)) toks=$(INITIAL_TOKENS),100000000uagstake ;; \
-			*) toks=1000000uag ;; \
-		esac; \
-		$(AGC) add-genesis-account `cat t1/$$port/ag-cosmos-helper-address` $$toks; \
 	done
-	$(AGC) gentx --keyring-backend=test --home-client=t1/$(BASE_PORT)/ag-cosmos-helper-statedir \
-		--name=ag-solo --amount=1000000uagstake
-	$(AGC) collect-gentxs
-	$(AGC) validate-genesis
-	../deployment/set-json.js ~/.ag-chain-cosmos/config/genesis.json --agoric-genesis-overrides
+	# Create the bootstrap account.
+	mkdir t1/bootstrap
+	$(AGCH) --home=t1/bootstrap keys add bootstrap --keyring-backend=test
+	$(AGCH) --home=t1/bootstrap keys show -a bootstrap --keyring-backend=test > t1/bootstrap-address
+	$(AGC) --home=t1/n0 add-genesis-account `cat t1/bootstrap-address` 100000000uagstake
+	# Create the (singleton) chain node.
+	$(AGC) --home=t1/n0 gentx --keyring-backend=test --home-client=t1/bootstrap --name=bootstrap --amount=1000000uagstake
+	$(AGC) --home=t1/n0 collect-gentxs
+	$(AGC) --home=t1/n0 validate-genesis
+	../deployment/set-json.js t1/n0/config/genesis.json --agoric-genesis-overrides
+	# Set the chain address in all the ag-solos.
 	$(MAKE) set-local-gci-ingress
-	@echo "ROLE=two_chain BOOT_ADDRESS=\`cat t1/$(BASE_PORT)/ag-cosmos-helper-address\` $(AGC) start"
-	@echo "(cd t1/$(BASE_PORT) && ../bin/ag-solo start --role=two_client)"
 
 scenario2-run-chain:
-	set -e; ba=; for acha in t1/*/ag-cosmos-helper-address; do \
-		ba="$$ba "`cat $$acha`; \
-	done; \
-	ROLE=two_chain BOOT_ADDRESS="$$ba" $(NODE_DEBUG) \
-	  `$(BREAK_CHAIN) && echo --inspect-brk` $(AGC) start --pruning=nothing
-scenario2-run-client:
+	ROLE=two_chain BOOT_ADDRESS=`cat t1/bootstrap-address` $(NODE_DEBUG) \
+	  `$(BREAK_CHAIN) && echo --inspect-brk` $(AGC) --home=t1/n0 start --pruning=nothing
+
+# Provision and start a client.
+scenario2-run-client: t1-provision-one t1-start-ag-solo
```

Comment:
Previously, `scenario2-setup` was "must-be-done-once", while `scenario2-run-client` was sort of idempotent. In particular, it was quite reasonable to SIGINT your client node and then re-run `make scenario2-run-client` to start it back up again, knowing it would pick up where it left off by re-running the transcripts.

It looks like this new `scenario2-run-client` won't behave that way, because it would re-run the `provision-one` message.

Should we maybe move the `t1-provision-one` out of `scenario2-run-client` and into `scenario2-setup`? Folks can run an extra `BASEPORT=xx make t1-provision-one` to spin up additional nodes.